### PR TITLE
Improve Meson

### LINKED ISSUE #91: Add appdata to be translated
Translating `appdata.xml` file will help people to know about your app in the AppCenter. For example mostly French users won't download the app if it's description is in English, 'cause they will think that the app isn't translated and won't take time to read if its not translated.

### PR DESCRIPTION
**Pull request by ryonakano**
I made another PR based on the `develop` branch.

## Changes Summary
- Remove unused constants
- Unify the number of indentation to 4 (for related files)
- Separate translation files between the app itself and metadata (fixes #91)
- Remove no longer used pot (the current one is `com.github.dahenson.agenda.pot`)
- Remove stale INSTALL (superseded by the latest installation instruction in `README.md`)
- README: Update installation instructions to Meson

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,16 +27,19 @@ Agenda is designed and built for [elementary OS](https://elementary.io), and is
 If you want to hack on Agenda, you'll need the following packages installed:
 * libgtk-3-dev
 * libgranite-dev
+* meson
 * valac
 
-Then do the `cmake` dance to build and install:
+Run `meson build` to configure the build environment. Change to the build directory and run `ninja` to build
```
Comment (dahenson) [inline README.md:33]: Thank you, this was on my TODO list! :+1:

File: po/extra/meson.build [hunk 2]
```diff
@@ -0,0 +1,4 @@
+i18n.gettext('extra',
+    args: ['--directory='+meson.source_root(), '--from-code=UTF-8'],
+    install: false
```
Comment (dahenson) [inline po/extra/meson.build:3]: I understand that this now includes the extra files, which we need for internationalization, but why is this set to `false`? I'm not as versed in meson lingo as I should be.

File: vapi/config.vapi [hunk 3]
```diff
@@ -1,7 +1,5 @@
 [CCode (cprefix = "", lower_case_cprefix = "", cheader_filename = "config.h")]
 namespace Build {
-	public const string DATADIR;
-	public const string PKGDATADIR;
-	public const string GETTEXT_PACKAGE;
-	public const string VERSION;
```
Comment (dahenson) [inline vapi/config.vapi:6]: To my knowledge, only the `PKGDATADIR` is called in the code. Can you tell me why we would keep `GETTEXT_PACKAGE` as well?

I would like to keep `VERSION` for a future CLI flag feature.

**Review by dahenson — CHANGES_REQUESTED** [review]
Thank you for cleaning this up! Just a few questions before I merge.

File: vapi/config.vapi (hunk 3 above)
Comment (ryonakano) [reply vapi/config.vapi:6]: I've once removed `GETTEXT_PACKAGE` in my previous "Add Meson" PR, but the CI complains its unexistence, although I've removed codes that uses it: https://travis-ci.org/dahenson/agenda/builds/612755020#L3463-L3469.

However, I'll remove this const again and see what happens :smile:

File: po/extra/meson.build (hunk 2 above)
Comment (ryonakano) [reply po/extra/meson.build:3]: The target of the `extra` domain are `.appdata.xml.in` and `.desktop.in` files. The former is used to see the app page in AppCenter and translations for it should be fetched from online, not from local. Also, the latter is compiled into `.desktop` file, whose contents look like this (the one for Agenda does not have translations ATM, so I quote the one of my app):

```
[Desktop Entry]
Name[es]=Reco
Name[fr]=Reco
Name[ja]=Reco
Name[lt]=Reco
Name[nl]=Reco
Name[pl]=Reco
Name=Reco
GenericName[es]=Aplicación para grabar audio
GenericName[fr]=Aplication d'enregistrement audio
GenericName[ja]=オーディオ録音アプリ
GenericName[lt]=Garso rašytuvo programa
GenericName[nl]=Audio-opnameapp
GenericName[pl]=Rejestrator dźwięku
GenericName=Audio Recording App
Comment[es]=Grabe charlas para recordar el contenido después
Comment[fr]=Enregistrez des discussions pour vous en rappeler le contenu plus tard
Comment[ja]=後で内容を思い出せるように会話を録音します
Comment[lt]=Įrašinėti pokalbius, siekiant vėliau prisiminti jų turinį
Comment[nl]=Neem gesprekken op om ze later terug te luisteren
Comment[pl]=Nagrywaj mowę, żeby pamiętać treści później
Comment=Record talks to remember the contents later
……
```

So, `.desktop` file itself includes translations for itself. Thus, we don't need to install translation files for the extra domain and do need `install: false` flag.

**Comment by ryonakano** [comment]
Also, could you merge `master` into `develop` to reflect [this change](https://github.com/dahenson/agenda/commit/2fc83fa26d94c34885d26f8e5e940baa11791b2d) so that we can pass the CI?

**Comment by dahenson** [comment]
@ryonakano I've merged master into develop. I still think it will fail to build, though. I think it's failing on the Loki build, and it seems like I might have to "release" all of the current changes and merge `develop` into `master` to get those issues to clear up. However, this is just a hunch and I haven't been comfortable until now that `develop` is stable enough to prepare for a new release.

**Review by dahenson — APPROVED** [review]

**Comment by ryonakano** [comment]
Thank you! I've merged the latest `develop` into this branch and triggered rebuilding but [the CI complains](https://travis-ci.org/dahenson/agenda/builds/624395932#L3466-L3469
) that the following line has no necessary argument:

```
../../src/Agenda.vala:44.22-44.40: error: 1 missing arguments for `void Agenda.AgendaWindow.new (Agenda.Agenda app)'
            window = new AgendaWindow ();
                     ^^^^^^^^^^^^^^^^^^^
Compilation failed: 1 error(s), 0 warning(s)
```

However, as far as I confirmed, that argument is there and building locally successes. I'm not really sure why the CI still fails. :woman_shrugging:

**Comment by dahenson** [comment]
Yes! This is exactly the issue I've been running into, and I think it has to do with attempting to build what's in master, since that's how the old code was structured. I suspect it might be an issue with how I understand Houston should operate, or an actual bug in Houston. I filed a bug and notified the elementary team, but haven't had any traction. The issue is here: https://github.com/elementary/houston/issues/744

**Comment by ryonakano** [comment]
@dahenson I enabled Travis CI in my forked repository and had some experiments: https://travis-ci.org/ryonakano/agenda/builds?utm_medium=notification&utm_source=github_status. And I found that **removing `deb-packaging-loki` branch fixes the CI.** Maybe some configuration in that branch would be the cause. If you want to keep support for Loki, we need to investigate what is the cause and fix it. However, I personally think we can stop providing Agenda for Loki rather than fixing it since Loki is the 2 versions before the latest release, Hera.

**Comment by dahenson** [comment]
Thanks again for working through this with me.

**Comment by ryonakano** [comment]
Thank you, too!! :tada:

**Pull request merged**